### ISSUE TITLE
Requests to webdav `/meta/`endpoint with valid fileId returns 404 Not Found

### ISSUE THREAD

**Issue by saw-jan** [owncloud/ocis#1978]
### Steps
1. create file
```bash
curl -X PUT -u admin:admin https://localhost:9200/remote.php/webdav/newFile.txt -v
```
2. get file-id from response header
```bash
...
< Oc-Fileid: MTI4NGQyMzgtYWE5Mi00MmNlLWJkYzQtMGIwMDAwMDA5MTU3Ojk5ZmJhYTEyLWI4NDItNGY0OC1iZDFiLTIyZmU0OTY2ZjFlYg==
...
```
3. Request with file-id
```bash
curl -X PROPFIND -u admin:admin \
https://localhost:9200/remote.php/dav/meta/MTI4NGQyMzgtYWE5Mi00MmNlLWJkYzQtMGIwMDAwMDA5MTU3Ojk5ZmJhYTEyLWI4NDItNGY0OC1iZDFiLTIyZmU0OTY2ZjFlYg== -v
```
**Actual Response**:
```xml
> PROPFIND /remote.php/dav/meta/MTI4NGQyMzgtYWE5Mi00MmNlLWJkYzQtMGIwMDAwMDA5MTU3Ojk5ZmJhYTEyLWI4NDItNGY0OC1iZDFiLTIyZmU0OTY2ZjFlYg== HTTP/1.1
> Host: localhost:9200
> Authorization: Basic YWRtaW46YWRtaW4=
> User-Agent: curl/7.68.0
> Accept: */*
> 
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< Access-Control-Allow-Origin: *
< Content-Length: 0
< Content-Security-Policy: default-src 'none';
< Date: Wed, 28 Apr 2021 05:37:01 GMT
< Vary: Origin
< X-Content-Type-Options: nosniff
< X-Download-Options: noopen
< X-Frame-Options: SAMEORIGIN
< X-Permitted-Cross-Domain-Policies: none
< X-Robots-Tag: none
< X-Xss-Protection: 1; mode=block
```

**Comment by ScharfViktor** [comment]
Method with **'/v'** works correct. @saw-jan maybe you missed it?

<img width="991" alt="Screenshot 2021-09-08 at 14 58 58" src="https://user-images.githubusercontent.com/84779829/132514042-cd6dad87-a218-4c14-b470-6db27fffe966.png">

**Comment by saw-jan** [comment]
@ScharfViktor Yeah, it works. Didn't know that I need to use `/v` after fileId.
So is `/v` mandatory?

**Comment by phil-davis** [comment]
@saw-jan please check against oC10. It will be good to know if `/v` is required there.

**Comment by saw-jan** [comment]
@phil-davis Yeah, I tried recently and in oC10, I can get status `207` and some response  without `/v`

**Comment by phil-davis** [comment]
@saw-jan please post examples of responses with and without `/v` against both oC10 and oCIS. Then we can see which have differences. After that we can look at the specs/documentation of what is required to work, and maybe we will need to ask about what the existing clients are actually doing, and then someone decide if there is anything that needs fixing/changing in oCIS or oC10.

**Comment by saw-jan** [comment]
Ok, sure
```xml
```

**Comment by saw-jan** [comment]
re-tested `oCIS 1.19.1`.
Now, creating file in oCIS gives `oC-Fileid` like `ddc2004c-0977-11eb-9d3f-a793888cd0f8!eef1649a-50b9-47aa-ade3-cd38524af8ea`

Creating file:
```
curl -XPUT -uadmin:admin https://host.docker.internal:9200/remote.php/webdav/file.txt -vk

< HTTP/1.1 201 Created
< Access-Control-Allow-Origin: *
< Content-Length: 0
< Content-Security-Policy: default-src 'none';
< Content-Type: text/plain
< Date: Wed, 06 Apr 2022 08:22:15 GMT
< Etag: "40720c43d1946666c74ee5aeecff4a64"
< Last-Modified: Wed, 06 Apr 2022 08:22:15 +0000
< Oc-Etag: "40720c43d1946666c74ee5aeecff4a64"
< Oc-Fileid: ddc2004c-0977-11eb-9d3f-a793888cd0f8!eef1649a-50b9-47aa-ade3-cd38524af8ea
< Vary: Origin
< X-Content-Type-Options: nosniff
< X-Download-Options: noopen
< X-Frame-Options: SAMEORIGIN
< X-Permitted-Cross-Domain-Policies: none
< X-Robots-Tag: none
< X-Xss-Protection: 1; mode=block
```

And `/meta/<oc-file-id>/v` won't work
```
curl -XPROPFIND -uadmin:admin https://host.docker.internal:9200/remote.php/dav/meta/ddc2004c-0977-11eb-9d3f-a793888cd0f850b9-47aa-ade3-cd38524af8ea/v -vk

< HTTP/1.1 404 Not Found
< Access-Control-Allow-Origin: *
< Content-Length: 14
< Content-Security-Policy: default-src 'none';
< Content-Type: text/plain; charset=utf-8
< Date: Wed, 06 Apr 2022 08:29:12 GMT
< Vary: Origin
< X-Content-Type-Options: nosniff
< X-Download-Options: noopen
< X-Frame-Options: SAMEORIGIN
< X-Permitted-Cross-Domain-Policies: none
< X-Robots-Tag: none
< X-Xss-Protection: 1; mode=block
```

**Comment by DeepDiver1975** [comment]
🤖 Automated issue triage by the Claude Code review agent.

**Closing as resolved / working-as-designed**, with evidence from the current codebase and acceptance tests.

The `/dav/meta/<fileId>` endpoint is **not** meant to be queried bare — it only exposes the versions collection under the `/v` sub-path. As @ScharfViktor demonstrated in this thread, `PROPFIND /dav/meta/<fileId>/v` works correctly in oCIS; only the bare path (no `/v`) returns 404.

This is now intentional and codified:

- **reva handler** (`internal/http/services/owncloud/ocdav/meta.go`): after parsing the resource ID, the handler routes only `head == "v"` to the versions handler; **any other sub-path (including an empty one, i.e. no `/v`) returns 404 by design**. An unparseable ID returns 400.
- **oCIS acceptance tests** (`tests/acceptance/features/coreApiVersions/fileVersions.feature`) lock this in: scenarios for the meta folder expect HTTP `400` or `404` for the bare `/dav/meta/<FILEID>` path and for invalid file IDs, while the helper (`tests/acceptance/TestHelpers/WebDavHelper.php`) always uses `dav/meta/<id>/v` for valid version access.
- The `/dav/meta` endpoint was additionally reworked/hardened in **oCIS 5.0.0** — see [cs3org/reva#4474](https://github.com/cs3org/reva/pull/4474) ("Make /dav/meta consistent") and [cs3org/reva#4432](https://github.com/cs3org/reva/pull/4432) ("Fix /dav/meta endpoint for shares").

Note also that the last reproduction in this thread (oCIS 1.19.1, 2022) used a **malformed fileId** — the `\!` separator and part of the ID were stripped (`...cd0f850b9...` instead of `...cd0f8\!eef1649a-50b9...`), which would itself fail ID parsing. With a correctly formed fileId and the required `/v` suffix, the endpoint responds as expected.

If you can still reproduce a 404 on a current oCIS release using a correctly formed fileId **with** the `/v` suffix, please **reopen** with the exact request/response and the oCIS version — thank you\!
